extensions/amp-analytics/0.1/visibility-manager.js

```diff
@@ -74,9 +75,10 @@ function createVisibilityManager(rootNode) {
   const ampdoc = Services.ampdoc(rootNode);
   const frame = getParentWindowFrameElement(rootNode);
   const embed = frame && getFriendlyIframeEmbedOptional(frame);
-  if (embed && frame && frame.ownerDocument) {
+  const frameRootNode = frame && rootNodeFor(frame);
+  if (embed && frameRootNode) {
     return new VisibilityManagerForEmbed(
-      provideVisibilityManager(frame.ownerDocument),
+      provideVisibilityManager(frameRootNode),
```

Comment:
Is the change to fix shadow doc only? If so, what's the current behavior for shadow using `frame.ownerDocument`.

Reply:
The chain here is top > shadow doc > fie. But as a general rule of thumb, `getRootNode` is a much safer API for most of cases. We're looking at how to get this into a more enforced world.

Reply:
I see. That makes sense. Didn't realize there's top > shadow doc > fie use case : )Thanks